### DISCUSSION TITLE
Shadowban

### DISCUSSION THREAD

**Discussion by NotAFile** [piqueserver/piqueserver#641] [Feature Requests]
This is a possibility that came up in a discussion with BR that we might want to explore.

Shadowbanning is the solution to users ban evading. It leads to them believing they are doing certain actions on the server, while in reality other users don't see those actions.

For example, if a griefer is shadowbanned, the server sends the client packets indicating the block has been removed, while not actually removing them from the map. The griefer will then go ahead and waste their time "destroying" the map.

The same could be done for aimbotters. Shadowbanned aimbotters would receive kill confirmations, but the shot player would still actually be alive.

**Comment by dgadelha** [comment]
That would be a really nice feature to have.
